src/Components/WebAssembly/WebAssembly.Authentication/src/Services/RemoteAuthenticationService.cs

```diff
@@ -165,6 +165,7 @@ public virtual async ValueTask<AccessTokenResult> RequestAccessToken()
     }
 
     /// <inheritdoc />
+    [DynamicDependency(JsonSerialized, typeof(AccessToken))]
```

Comment:
Does it matter where exactly we add the `DynamicDependency`?

Reply:
The way to think about it is - if the trimmer decides not to trim away `RequestAccessToken` because it's called by user code directly or indirectly, then the types declared as dynamic dependencies do not get trimmed. Usually you want to put it close to where the referenced type gets used so you're not rooting more things than you need to. I'm guessing in this case `InternalAccessTokenResult` references `AccessToken`?

Reply:
> I'm guessing in this case `InternalAccessTokenResult` references `AccessToken`?

Exactly, that's what's in the stack trace. But I wasn't sure if it needed to be on a method to work. Thanks for the clarification!

```
   at System.Text.Json.Serialization.Converters.ObjectDefaultConverter`1[[Microsoft.AspNetCore.Components.WebAssembly.Authentication.InternalAccessTokenResult, Microsoft.AspNetCore.Components.WebAssembly.Authentication, Version=7.0.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60]].OnTryRead(Utf8JsonReader& , Type , JsonSerializerOptions , ReadStack& , InternalAccessTokenResult& )
   at System.Text.Json.Serialization.JsonConverter`1[[Microsoft.AspNetCore.Components.WebAssembly.Authentication.InternalAccessTokenResult, Microsoft.AspNetCore.Components.WebAssembly.Authentication, Version=7.0.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60]].TryRead(Utf8JsonReader& , Type , JsonSerializerOptions , ReadStack& , InternalAccessTokenResult& )
   at System.Text.Json.Serialization.JsonConverter`1[[Microsoft.AspNetCore.Components.WebAssembly.Authentication.InternalAccessTokenResult, Microsoft.AspNetCore.Components.WebAssembly.Authentication, Version=7.0.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60]].ReadCore(Utf8JsonReader& , JsonSerializerOptions , ReadStack& )
   at System.Text.Json.Serialization.JsonConverter`1[[Microsoft.AspNetCore.Components.WebAssembly.Authentication.InternalAccessTokenResult, Microsoft.AspNetCore.Components.WebAssembly.Authentication, Version=7.0.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60]].ReadCoreAsObject(Utf8JsonReader& , JsonSerializerOptions , ReadStack& )
   at System.Text.Json.JsonSerializer.ReadCore[Object](JsonConverter , Utf8JsonReader& , JsonSerializerOptions , ReadStack& )
```

Reply:
> But I wasn't sure if it needed to be on a method to work.

Looks like that is in fact the case:

> Attribute 'DynamicDependency' is not valid on this declaration type. It is only valid on 'constructor, method, field' declarations.

Reply:
Reverted back, any objections?

https://github.com/dotnet/aspnetcore/pull/40274/commits/827bd4de55c4722e6da777584f06d02567173bb9